cilium-cli/connectivity/tests/egressgateway.go

```diff
@@ -110,14 +162,17 @@ func (s *egressGateway) Run(ctx context.Context, t *check.Test) {
 	i := 0
 	for _, client := range ct.ClientPods() {
 		for _, dst := range ct.HostNetNSPodsByNode() {
-			t.NewAction(s, fmt.Sprintf("ping-%d", i), &client, &dst, features.IPFamilyV4).Run(func(a *check.Action) {
-				a.ExecInPod(ctx, ct.PingCommand(dst, features.IPFamilyV4))
+			t.ForEachIPFamily(func(ipFam features.IPFamily) {
+				t.NewAction(s, fmt.Sprintf("ping-%s-%d", ipFam, i), &client, &dst, ipFam).Run(func(a *check.Action) {
```

Comment:
Just to confirm, do we always run the tests that are unrelated to the EGW functionality without checking `ipv6Enabled`?

Reply:
Yes. ForEachIPFamily checks internally if the cluster supports IPv6, and those test should succeed regardless of EGW enabled. Overall we might want to migrate these test-cases to a more general test suite.